src/cluster.h

```diff
@@ -245,11 +246,38 @@ typedef struct {
     unsigned char bulk_data[3]; /* 3 bytes just as placeholder. */
 } clusterMsgModule;
 
+/* The cluster supports optional extension messages that can be sent
+ * along with ping/pong/meet messages to give additional info in a 
+ * consistent manner. */
+typedef enum {
+    CLUSTERMSG_EXT_TYPE_HOSTNAME,
+} clusterMsgPingtypes; 
+
+/* Helper function for making sure extensions are eight byte aligned. */
+#define EIGHT_BYTE_ALIGN(size) ((((size) + 7) / 8) * 8)
+
+typedef struct {
+    char hostname[1]; /* The announced hostname, ends with \0. */
+} clusterMsgPingExtHostname;
+
+typedef struct {
+    uint32_t length; /* Total length of this extension message (including this header) */
```

Comment:
The idea was that we wanted the start of the extension data to still be 8 byte aligned, which means we don't get any benefit by shrinking this since we still need to pad before the start of the data. The data needs to be 8 byte aligned since we want the start of a future extension to also be 8 byte aligned.

Reply:
Understand the alignment requirement. 

I was trying to see if we could save a few more bytes here but if the new "hostname" extension can be made optional then this optimization is moot. I think this is good. Feel free to resolve/close this comment.